constraints/miden-vm/stack.air

```diff
@@ -396,320 +493,396 @@ ev is_unchanged(main: [values[4]]):
     enf v' = v for v in values
 
 
+# Enforces that the provided columns must be binary.
+ev is_binary(main: [a]):
+    enf a^2 = a
+
+
 ev range_check_16bit()
 
 
 ### Stack Air Constraints #########################################################################
 
 # Enforces the constraints on the stack.
 # TODO: add docs for columns
-# ev stack_constraints(main: [stack[16], bookkeeping[2], helper], aux: [op_bits[7], extra, hasher5]):
+ev stack_constraints(main: [stack[16], bookkeeping[2], h0], aux: [op_bits[7], extra, hasher5, fmp, helper[3], clk, b_chip]):
+    let op_flags = compute_op_flags(op_bits, extra)
+    match enf:
+        noop([stack]) when op_flags[0]
+        eqz([stack, h0]) when op_flags[1]
+        neg([stack]) when op_flags[2]
+        inv([stack]) when op_flags[3]
```

Comment:
I would try to align `when` parts as much as possible to make this easier to read.